test/user.test.ts

```diff
@@ -0,0 +1,182 @@
+      password: TEST_AUTH_PASSWORD,
+    });
+  });
+
+  it("sends a request to delete user's account", async () => {
+    const { data } = await saleor.user.accountRequestDeletion(API_URI);
+    expect(data?.accountRequestDeletion?.errors).toHaveLength(0);
+  });
+
+  it("sends request to change user's email", async () => {
+    const { data } = await saleor.user.requestEmailChange({
+      newEmail: `register+${Date.now().toString()}@example.com`,
+      password: TEST_AUTH_PASSWORD,
+      redirectUrl: API_URI,
+    });
+    expect(data?.requestEmailChange?.errors).toHaveLength(0);
+  });
+
+  it("updates the user's first name", async () => {
+    const { data } = await saleor.user.updateAccount({
+      input: {
+        firstName: "",
+      },
+    });
+    expect(data?.accountUpdate?.user?.firstName).toBe("");
+    expect(data?.accountUpdate?.errors).toHaveLength(0);
+  });
+
+  it("creates the user's account address", async () => {
+    const { data } = await saleor.user.createAccountAddress({
+      input: testAddress,
+    });
+
+    if (data?.accountAddressCreate?.user?.addresses?.length) {
+      expect(
+        data?.accountAddressCreate?.user?.addresses[
+          data?.accountAddressCreate?.user?.addresses?.length - 1
+        ].firstName === testAddress.firstName
+      );
+    }
+    expect(data?.accountAddressCreate?.errors).toHaveLength(0);
+  });
+
+  it("updates the user's account address", async () => {
+    const cache = client.readQuery({
+      query: USER,
+    });
+    const index = cache.me.addresses.length - 1;
+    const newTestName = "New test name";
+    const { data } = await saleor.user.updateAccountAddress({
+      id: cache.me.addresses[index].id,
+      input: {
+        ...testAddress,
+        firstName: newTestName,
+      },
+    });
+    const updatedCachedUser = client.readQuery({
+      query: USER,
+    });
+    if (data?.accountAddressUpdate?.user?.addresses?.length) {
+      expect(
+        data?.accountAddressUpdate?.user?.addresses[index]?.firstName
+      ).toBe(newTestName);
+      expect(updatedCachedUser?.me?.addresses[index]?.firstName).toBe(
+        newTestName
+      );
+    }
+    expect(data?.accountAddressUpdate?.errors).toHaveLength(0);
+  });
+
+  it("sets address as a default billing address", async () => {
+    const cache = client.readQuery({
+      query: USER,
+    });
+    const index = cache.me.addresses.length - 1;
+    const addressId = cache.me.addresses[index].id;
+    const oldAddress = cache.me.defaultBillingAddress;
+    const { data } = await saleor.user.setAccountDefaultAddress({
+      id: addressId,
+      type: "BILLING",
+    });
+    expect(
+      data?.accountSetDefaultAddress?.user?.defaultBillingAddress?.id
+    ).toBe(addressId);
+    expect(data?.accountSetDefaultAddress?.errors).toHaveLength(0);
+
+    // Set back default testing address
+    const { data: revertedData } = await saleor.user.setAccountDefaultAddress({
+      id: oldAddress.id,
```

Comment:
Getting `TypeError: Cannot read property 'id' of null` from this line.
BTW why are we need to revert old address?

Reply:
No `TypeError` by me, the test was recorded fine. I wanted to go back to the current test address, not to set the dummy address.

Reply:
I got it why it fails - shared account no longer has any address

Reply:
should we worry about this? We already have recorded a test recording.

Reply:
We should - tests must be as independent as possible. 
The first record is not an issue in most cases, problems arise when the next person changes "one little thing" and has to refresh recordings. And then has to fight with some assumed server state. 
Solution: you can create a new user in the test setup and run mutations using its account. 
Also in this test you can create 2 address and check if switching works as intended without depending on some server state.